test/vestingscheme.js

```diff
@@ -115,6 +115,36 @@ contract('VestingScheme', function(accounts) {
           }
          });
 
+           it("execute proposeVestingAgreement- ProposedVestedAgreement supplies proposalId ", async function() {
+             var testSetup = await setup(accounts);
+
+
+             var tx = await testSetup.vestingScheme.proposeVestingAgreement(accounts[0],
+                                                                            accounts[1],
+                                                                            web3.eth.blockNumber,
+                                                                            15,
+                                                                            2,
+                                                                            3,
+                                                                            11,
+                                                                            0,
+                                                                            [],
+                                                                            testSetup.org.avatar.address);
+            //Vote with reputation to trigger execution
+             var proposalId = await helpers.getValueFromLogs(tx, '_proposalId',1);
+             await testSetup.vestingSchemeParams.votingMachine.absoluteVote.vote(proposalId,1,{from:accounts[2]});
+            const found = await new Promise((resolve) => {
+                testSetup.vestingScheme.ProposedVestedAgreement({_proposalId: proposalId}, {fromBlock: 0})
+                    .get((err,events) => {
```

Comment:
are you assuming here that there will be only one event (one test on the same test blockchain instance ) from block 0  ? Not sure if it is enough.

Reply:
Good idea,  but note that I am filtering the events by the proposalId, which is unique.  So there should only be a single event returned.  Thus I think this is OK for the case you described.

Reply:
why you are not getting the event in the normal standard way?

Reply:
By "normal standard way", I assume you mean from the tx log after voting?

It is because `ProposedVestedAgreement` doesn't appear in the tx.logs.  Only `VoteProposal` and `ExecuteProposal` appear there.  Yet `ProposedVestedAgreement` *is* being fired.

Reply:
How about reading the block number of the tx and explicitly set the fromBlock and toBlock with that ?

Reply:
@dkent600 is that been fixed?

Reply:
yes